src/main/java/yalter/mousetweaks/MouseState.java

```diff
@@ -0,0 +1,73 @@
+package yalter.mousetweaks;
+
+import java.util.EnumSet;
+
+import com.google.common.base.MoreObjects;
+import org.lwjgl.input.Mouse;
+
+public class MouseState {
+	private final EnumSet<MouseButton> pressedButtons = EnumSet.noneOf(MouseButton.class);
+	private int scrollAmount = 0;
+
+	public MouseState() {}
+
+	/**
+	 * Update the current mouse state.
+	 */
+	public void update() {
+		MouseButton eventButton = getEventButton();
```

Comment:
Are we guaranteed to get multiple update calls from Forge even if multiple buttons were pressed at once? Even if so, there doesn't seem to be any special handling in the LiteLoader code—think pressing or releasing LMB and RMB at the same time.

Reply:
Vanilla consumes the mouse buttons one at a time, and Forge fires an event for each one.

I tested with pressing LMB and RMB at the same time, this code handles it and `Main` uses something like `if (right) elseif (left) else` so it will not try to do 2 things at once, it will just use the right button if both are pressed.

Reply:
I guess this is more of a concern for release: press RMB, then press LMB, then release both at once. With a low enough FPS this can easily happen. Will the LiteLoader code work correctly in this case or will the mod think one of the buttons got stuck pressed?

Reply:
I think you're right and it could be an issue, I'll take a look.

Reply:
I pushed a fix, it will check `Mouse.isButtonDown` to confirm anything that `MouseState` thinks is pressed.
This way `MouseState` can still prevent using buttons from previous guis, and we get real confirmation that it doesn't have some "stuck" button.